### PR TITLE
Meging Dev changes into Master

### PR DESCRIPTION
**Pull request by LordOfPolls**
## What type of pull request is this?

- [x] Non-breaking code change
- [ ] Breaking code change
- [ ] Documentation change/addition 

## Description

Merges `Dev` branches changes into `master`

## Changes

- MyPy/Typehint fixes from @silasary 
- Emoji object can now parse discord emoji strings
- Added Dinteractions V3 subcommand decorator 
- Various doc fixes from @Kigstn 
- Add support for member timeouts from @Wolfhound905 
- Fix `fetch_members` unable to access websocket
- Warn when `fetch_members` is used without the members intent
- Fix unnecessary caching of scopes for app commands
- Potentially fix gateway none error
- Prevent useless errors being rasied by Auto-Defer
- Resume event now sets the ready signal 

## Checklist

- [x] I've formatted my code with [Black](https://black.readthedocs.io/en/stable/)
- [x] I've ensured my code works on `Python 3.10.x`
- [x] I've tested my code

### CODE REVIEW

**Comment by LordOfPolls** [comment]
As this is merging into master, we require 2 approving reviews

**Review by AstreaTSS — APPROVED** [review]
LGTM

File: dis_snek/models/discord_objects/user.py [hunk 1]
```diff
@@ -191,6 +194,11 @@ class Member(DiscordObject, _SendDMMixin):
         default=None, metadata=docs("Whether the user has **not** passed guild's membership screening requirements")
     )
     guild_avatar: "Asset" = field(default=None, metadata=docs("The user's guild avatar"))
+    communication_disabled_until: Optional["Timestamp"] = field(
+        default=None,
+        converter=optional_c(timestamp_converter),
+        metadata=docs("wWen a member's timeout will expire, `None` or a time in the past if the user is not timed out"),
```
Comment (KAJdev) [inline dis_snek/models/discord_objects/user.py]: Typo in docstrings
```suggestion
        metadata=docs("When a member's timeout will expire, `None` or a time in the past if the user is not timed out"),
```

File: dis_snek/models/discord_objects/user.py [hunk 2]
```diff
@@ -423,6 +431,28 @@ def has_role(self, *roles: Union[Snowflake_Type, Role]) -> bool:
...
 
+    async def timeout(
+        self, communication_disabled_until: Union["Timestamp", datetime, int, float, str, None], reason: str = MISSING
+    ):
+        """
+        Disable a members communication for a given time.
+
+        Args:
+            communication_disabled_until: The time until the user can communicate again
+            reson: The reason for this timeout
```
Comment (KAJdev) [inline dis_snek/models/discord_objects/user.py]:
```suggestion
            reason: The reason for this timeout
```
fix another parameter typo

**Review by KAJdev — APPROVED** [review]
👍

**Review by zevaryx — APPROVED** [review]
Looks good

**Review by artem30801 — APPROVED** [review]
LGTM

**Pull request merged**